src/lib/utils/scales-utils.js

```diff
@@ -293,7 +293,8 @@ function _computeRightDomainAdjustment(values) {
  */
 function _computeScaleDistance(values, domain, index, scaleFn) {
```

Comment:
I suggest renaming index to `shortestDistIndex` or `bestDistIndex` since as it reads right now, it looks as if this scale distance is called for all indexes.